.pre-commit-config.yaml

```diff
@@ -0,0 +1,40 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.6.0
+    hooks:
+      - id: trailing-whitespace
+        exclude: ^pineappl_cli/tests/
+      - id: end-of-file-fixer
+      - id: check-merge-conflict
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    # A fast Python linter and code formatter. See
+    # https://docs.astral.sh/ruff/ for more details.
+    rev: v0.6.5
+    hooks:
+      - id: ruff
+        args: [--fix]
+        exclude: ^pineappl_cli/src/
+      - id: ruff-format
+        args: [--line-length=80]
+  - repo: local
+    hooks:
+    - id: fmt
+      name: cargo fmt
+      description: Format Rust files with cargo fmt.
+      entry: cargo fmt --
+      language: system
+      files: ^pineappl\S*\/.*\.rs$
+      args: []
+    - id: check
```

Comment:
In eko we have `cargo clippy` but not `cargo check` - which one is better? will either `fmt` or `clippy` run `check` along the way?

Reply:
They are different: `cargo check` is essentially running the compiler, without generating code. `cargo clippy` is statically analyzing the code (i.e. the equivalent of Pylint).

There is a non-negligible change that `cargo clippy` is a superset of `cargo check`, but it's also possible that they only partially overlap. But this you can search for (or even try: run `cargo clippy` on something that `cargo check` was flagging, and see if it's complaining - even a missing `;` or unmatched `{`).
The opposite is definitely impossible: `cargo clippy` may do strictly *more*, but certainly not strictly *less*.

Reply:
> They are different: cargo check is essentially running the compiler, without generating code. cargo clippy is statically analyzing the code (i.e. the equivalent of Pylint).

Yes, exactly. I did not include `clippy` before because it is very noisy but there is actually some ways to suppress warnings and only show errors.

Reply:
Well: that's certainly an option, but another one is just implementing its suggestions :P

Reply:
Yes, I agree! I will address this soon https://github.com/NNPDF/pineappl/issues/314 😉